### ISSUE TITLE
SIGSEGV during kpt live install-resource-group

### ISSUE THREAD

**Issue by apeabody** [GoogleContainerTools/kpt#3892]
SIGSEGV observed during `kpt live install-resource-group` with `1.0.0-beta.29` and `1.0.0-beta.27` binary from GH.  Issue not observed with `1.0.0-beta.20` (or earlier) binary from GH.

```
fatal error: unexpected signal during runtime execution
        [signal SIGSEGV: segmentation violation code=0x1 addr=0x27020 pc=0x27020]
        
        runtime stack:
        runtime.throw({0x294a34f?, 0x586a438?})
        	/usr/local/go/src/runtime/panic.go:1047 +0x5d fp=0x7fffbbbb2bd8 sp=0x7fffbbbb2ba8 pc=0x48347d
        runtime.sigpanic()
        	/usr/local/go/src/runtime/signal_unix.go:819 +0x369 fp=0x7fffbbbb2c28 sp=0x7fffbbbb2bd8 pc=0x499a29
        
        goroutine 54 [syscall]:
        runtime.cgocall(0x1a106d0, 0xc000086d98)
        	/usr/local/go/src/runtime/cgocall.go:158 +0x5c fp=0xc000086d70 sp=0xc000086d38 pc=0x44f7bc
        net._C2func_getaddrinfo(0xc0014aa040, 0x0, 0xc000d06d20, 0xc000fd0a28)
        	_cgo_gotypes.go:94 +0x56 fp=0xc000086d98 sp=0xc000086d70 pc=0x666516
        net.cgoLookupIPCNAME.func1({0xc0014aa040, 0x0?, 0x0?}, 0xc001463e50?, 0x66cebb?)
        	/usr/local/go/src/net/cgo_unix.go:160 +0x9f fp=0xc000086df0 sp=0xc000086d98 pc=0x66825f
        net.cgoLookupIPCNAME({0x28fdb9d, 0x3}, {0xc001463e50, 0x9})
        	/usr/local/go/src/net/cgo_unix.go:160 +0x173 fp=0xc000086f38 sp=0xc000086df0 pc=0x667ab3
        net.cgoIPLookup(0x3f6de30?, {0x28fdb9d?, 0xc001463e60?}, {0xc001463e50?, 0xc0000f1970?})
        	/usr/local/go/src/net/cgo_unix.go:217 +0x3b fp=0xc000086fa8 sp=0xc000086f38 pc=0x66831b
        net.cgoLookupIP.func1()
        	/usr/local/go/src/net/cgo_unix.go:227 +0x36 fp=0xc000086fe0 sp=0xc000086fa8 pc=0x668756
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc000086fe8 sp=0xc000086fe0 pc=0x4b5a01
        created by net.cgoLookupIP
        	/usr/local/go/src/net/cgo_unix.go:227 +0x12a
        
        goroutine 1 [select]:
        runtime.gopark(0xc002341450?, 0x4?, 0x18?, 0x0?, 0xc0023412f8?)
        	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc002341150 sp=0xc002341130 pc=0x4863b6
        runtime.selectgo(0xc002341450, 0xc0023412f0, 0xc001622cf0?, 0x0, 0xc?, 0x1)
        	/usr/local/go/src/runtime/select.go:328 +0x7bc fp=0xc002341290 sp=0xc002341150 pc=0x495abc
        net/http.(*Transport).getConn(0xc0018ea640, 0xc001a52e00, {{}, 0xc00191cfc0, {0xc001acbc80, 0x5}, {0xc001463e10, 0xe}, 0x0})
        	/usr/local/go/src/net/http/transport.go:1376 +0x5c6 fp=0xc0023414a8 sp=0xc002341290 pc=0x7f3de6
        net/http.(*Transport).roundTrip(0xc0018ea640, 0xc001ad9200)
        	/usr/local/go/src/net/http/transport.go:582 +0x75e fp=0xc0023416d0 sp=0xc0023414a8 pc=0x7ef7be
        net/http.(*Transport).RoundTrip(0x2cd4fb0?, 0xc0018cf620?)
        	/usr/local/go/src/net/http/roundtrip.go:17 +0x19 fp=0xc0023416f0 sp=0xc0023416d0 pc=0x7e32f9
        sigs.k8s.io/cli-utils/pkg/flowcontrol.IsEnabled({0x2cd4fb0, 0xc0018cf620}, 0xc001af8d80)
        	/root/go/pkg/mod/sigs.k8s.io/cli-utils@v0.34.0/pkg/flowcontrol/flowcontrol.go:64 +0x33f fp=0xc0023417c0 sp=0xc0023416f0 pc=0x1974c5f
        github.com/GoogleContainerTools/kpt/commands/util.UpdateQPS.func1(0xc001af8d80)
        	/go/src/github.com/GoogleContainerTools/kpt/commands/util/factory.go:64 +0x8b fp=0xc002341870 sp=0xc0023417c0 pc=0x197522b
        k8s.io/cli-runtime/pkg/genericclioptions.(*ConfigFlags).ToRESTConfig(0xc001a48500)
        	/root/go/pkg/mod/k8s.io/cli-runtime@v0.25.3/pkg/genericclioptions/config_flags.go:136 +0x42 fp=0xc002341888 sp=0xc002341870 pc=0x14b4722
        k8s.io/cli-runtime/pkg/genericclioptions.(*ConfigFlags).toDiscoveryClient(0xc001a48500)
        	/root/go/pkg/mod/k8s.io/cli-runtime@v0.25.3/pkg/genericclioptions/config_flags.go:266 +0x33 fp=0xc002341930 sp=0xc002341888 pc=0x14b5313
        k8s.io/cli-runtime/pkg/genericclioptions.(*ConfigFlags).toPersistentDiscoveryClient(0xc001a48500)
        	/root/go/pkg/mod/k8s.io/cli-runtime@v0.25.3/pkg/genericclioptions/config_flags.go:256 +0xa7 fp=0xc002341990 sp=0xc002341930 pc=0x14b5167
        k8s.io/cli-runtime/pkg/genericclioptions.(*ConfigFlags).ToDiscoveryClient(0xc00007cf00?)
        	/root/go/pkg/mod/k8s.io/cli-runtime@v0.25.3/pkg/genericclioptions/config_flags.go:246 +0x25 fp=0xc0023419a8 sp=0xc002341990 pc=0x14b5085
        k8s.io/cli-runtime/pkg/genericclioptions.(*ConfigFlags).toRESTMapper(0xc54d4c?)
        	/root/go/pkg/mod/k8s.io/cli-runtime@v0.25.3/pkg/genericclioptions/config_flags.go:324 +0x1d fp=0xc002341a00 sp=0xc0023419a8 pc=0x14b585d
        k8s.io/cli-runtime/pkg/genericclioptions.(*ConfigFlags).toPersistentRESTMapper(0xc001a48500)
        	/root/go/pkg/mod/k8s.io/cli-runtime@v0.25.3/pkg/genericclioptions/config_flags.go:314 +0xa7 fp=0xc002341a60 sp=0xc002341a00 pc=0x14b56c7
        k8s.io/cli-runtime/pkg/genericclioptions.(*ConfigFlags).ToRESTMapper(0x532384?)
        	/root/go/pkg/mod/k8s.io/cli-runtime@v0.25.3/pkg/genericclioptions/config_flags.go:304 +0x25 fp=0xc002341a78 sp=0xc002341a60 pc=0x14b55e5
        github.com/GoogleContainerTools/kpt/internal/util/cfgflags.(*UserAgentKubeConfigFlags).ToRESTMapper(0x0?)
        	/go/src/github.com/GoogleContainerTools/kpt/internal/util/cfgflags/useragent.go:46 +0x22 fp=0xc002341a90 sp=0xc002341a78 pc=0x1974822
        k8s.io/kubectl/pkg/cmd/util.(*factoryImpl).ToRESTMapper(0x29?)
        	/root/go/pkg/mod/k8s.io/kubectl@v0.25.3/pkg/cmd/util/factory_client_access.go:66 +0x22 fp=0xc002341aa8 sp=0xc002341a90 pc=0x17046a2
        github.com/GoogleContainerTools/kpt/pkg/status.NewStatusPoller({0x2ce0aa0, 0xc0013bbfc0})
        	/go/src/github.com/GoogleContainerTools/kpt/pkg/status/poller.go:29 +0x3f fp=0xc002341b38 sp=0xc002341aa8 pc=0x189bcbf
        github.com/GoogleContainerTools/kpt/pkg/live.(*ResourceGroupInstaller).InstallRG(0xc002341c28, {0x2cd4fe8, 0xc000c5a7b0})
        	/go/src/github.com/GoogleContainerTools/kpt/pkg/live/inventoryrg.go:422 +0x50 fp=0xc002341be0 sp=0xc002341b38 pc=0x18a3b50
        github.com/GoogleContainerTools/kpt/commands/live/installrg.(*Runner).runE(0xc0018cee40, 0x0?, {0x3fa9bc0?, 0x0?, 0x0?})
        	/go/src/github.com/GoogleContainerTools/kpt/commands/live/installrg/cmdinstallrg.go:80 +0x97 fp=0xc002341c48 sp=0xc002341be0 pc=0x19c3537
        github.com/GoogleContainerTools/kpt/commands/live/installrg.(*Runner).runE-fm(0xc001a68600?, {0x3fa9bc0?, 0x0?, 0x0?})
        	<autogenerated>:1 +0x3a fp=0xc002341c80 sp=0xc002341c48 pc=0x19c367a
        github.com/spf13/cobra.(*Command).execute(0xc001a68600, {0x3fa9bc0, 0x0, 0x0})
        	/root/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862 fp=0xc002341db8 sp=0xc002341c80 pc=0x1482da2
        github.com/spf13/cobra.(*Command).ExecuteC(0xc0000dac00)
        	/root/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd fp=0xc002341e70 sp=0xc002341db8 pc=0x148369d
        github.com/spf13/cobra.(*Command).Execute(...)
        	/root/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
        k8s.io/component-base/cli.run(0xc0000dac00)
        	/root/go/pkg/mod/k8s.io/component-base@v0.25.3/cli/run.go:146 +0x317 fp=0xc002341f30 sp=0xc002341e70 pc=0x1a0de37
        k8s.io/component-base/cli.RunNoErrOutput(...)
        	/root/go/pkg/mod/k8s.io/component-base@v0.25.3/cli/run.go:84
        main.runMain()
        	/go/src/github.com/GoogleContainerTools/kpt/main.go:62 +0x45 fp=0xc002341f68 sp=0xc002341f30 pc=0x1a0e105
        main.main()
        	/go/src/github.com/GoogleContainerTools/kpt/main.go:46 +0x19 fp=0xc002341f80 sp=0xc002341f68 pc=0x1a0e099
        runtime.main()
        	/usr/local/go/src/runtime/proc.go:250 +0x212 fp=0xc002341fe0 sp=0xc002341f80 pc=0x485ff2
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc002341fe8 sp=0xc002341fe0 pc=0x4b5a01
        
        goroutine 2 [force gc (idle)]:
        runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000088fb0 sp=0xc000088f90 pc=0x4863b6
        runtime.goparkunlock(...)
        	/usr/local/go/src/runtime/proc.go:369
        runtime.forcegchelper()
        	/usr/local/go/src/runtime/proc.go:302 +0xad fp=0xc000088fe0 sp=0xc000088fb0 pc=0x48624d
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc000088fe8 sp=0xc000088fe0 pc=0x4b5a01
        created by runtime.init.6
        	/usr/local/go/src/runtime/proc.go:290 +0x25
        
        goroutine 3 [GC sweep wait]:
        runtime.gopark(0x3f6ec01?, 0x0?, 0x0?, 0x0?, 0x0?)
        	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000089790 sp=0xc000089770 pc=0x4863b6
        runtime.goparkunlock(...)
        	/usr/local/go/src/runtime/proc.go:369
        runtime.bgsweep(0x0?)
        	/usr/local/go/src/runtime/mgcsweep.go:297 +0xd7 fp=0xc0000897c8 sp=0xc000089790 pc=0x4711f7
        runtime.gcenable.func1()
        	/usr/local/go/src/runtime/mgc.go:178 +0x26 fp=0xc0000897e0 sp=0xc0000897c8 pc=0x465d86
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc0000897e8 sp=0xc0000897e0 pc=0x4b5a01
        created by runtime.gcenable
        	/usr/local/go/src/runtime/mgc.go:178 +0x6b
        
        goroutine 4 [GC scavenge wait]:
        runtime.gopark(0x3f70340?, 0x2caf9a0?, 0x0?, 0x0?, 0x0?)
        	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000089f70 sp=0xc000089f50 pc=0x4863b6
        runtime.goparkunlock(...)
        	/usr/local/go/src/runtime/proc.go:369
        runtime.(*scavengerState).park(0x3f70340)
        	/usr/local/go/src/runtime/mgcscavenge.go:389 +0x53 fp=0xc000089fa0 sp=0xc000089f70 pc=0x46f1d3
        runtime.bgscavenge(0x0?)
        	/usr/local/go/src/runtime/mgcscavenge.go:622 +0x65 fp=0xc000089fc8 sp=0xc000089fa0 pc=0x46f7e5
        runtime.gcenable.func2()
        	/usr/local/go/src/runtime/mgc.go:179 +0x26 fp=0xc000089fe0 sp=0xc000089fc8 pc=0x465d26
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc000089fe8 sp=0xc000089fe0 pc=0x4b5a01
        created by runtime.gcenable
        	/usr/local/go/src/runtime/mgc.go:179 +0xaa
        
        goroutine 18 [finalizer wait]:
        runtime.gopark(0x3f71800?, 0xc000007860?, 0x0?, 0x0?, 0xc000088770?)
        	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000088628 sp=0xc000088608 pc=0x4863b6
        runtime.goparkunlock(...)
        	/usr/local/go/src/runtime/proc.go:369
        runtime.runfinq()
        	/usr/local/go/src/runtime/mfinal.go:180 +0x10f fp=0xc0000887e0 sp=0xc000088628 pc=0x464e8f
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc0000887e8 sp=0xc0000887e0 pc=0x4b5a01
        created by runtime.createfing
        	/usr/local/go/src/runtime/mfinal.go:157 +0x45
        
        goroutine 19 [GC worker (idle)]:
        runtime.gopark(0x62a8d9b8f136?, 0x0?, 0x0?, 0x0?, 0x0?)
        	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000084750 sp=0xc000084730 pc=0x4863b6
        runtime.gcBgMarkWorker()
        	/usr/local/go/src/runtime/mgc.go:1235 +0xf1 fp=0xc0000847e0 sp=0xc000084750 pc=0x467ed1
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc0000847e8 sp=0xc0000847e0 pc=0x4b5a01
        created by runtime.gcBgMarkStartWorkers
        	/usr/local/go/src/runtime/mgc.go:1159 +0x25
        
        goroutine 20 [GC worker (idle)]:
        runtime.gopark(0x62a8df99eb84?, 0x3?, 0x15?, 0x97?, 0x0?)
        	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000084f50 sp=0xc000084f30 pc=0x4863b6
        runtime.gcBgMarkWorker()
        	/usr/local/go/src/runtime/mgc.go:1235 +0xf1 fp=0xc000084fe0 sp=0xc000084f50 pc=0x467ed1
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc000084fe8 sp=0xc000084fe0 pc=0x4b5a01
        created by runtime.gcBgMarkStartWorkers
        	/usr/local/go/src/runtime/mgc.go:1159 +0x25
        
        goroutine 5 [GC worker (idle)]:
        runtime.gopark(0x2ad8f30?, 0xc00010e000?, 0x18?, 0x14?, 0x0?)
        	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc00008a750 sp=0xc00008a730 pc=0x4863b6
        runtime.gcBgMarkWorker()
        	/usr/local/go/src/runtime/mgc.go:1235 +0xf1 fp=0xc00008a7e0 sp=0xc00008a750 pc=0x467ed1
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc00008a7e8 sp=0xc00008a7e0 pc=0x4b5a01
        created by runtime.gcBgMarkStartWorkers
        	/usr/local/go/src/runtime/mgc.go:1159 +0x25
        
        goroutine 34 [GC worker (idle)]:
        runtime.gopark(0x62a8df99e511?, 0xc0003d34c0?, 0x18?, 0x14?, 0x0?)
        	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000114750 sp=0xc000114730 pc=0x4863b6
        runtime.gcBgMarkWorker()
        	/usr/local/go/src/runtime/mgc.go:1235 +0xf1 fp=0xc0001147e0 sp=0xc000114750 pc=0x467ed1
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc0001147e8 sp=0xc0001147e0 pc=0x4b5a01
        created by runtime.gcBgMarkStartWorkers
        	/usr/local/go/src/runtime/mgc.go:1159 +0x25
        
        goroutine 50 [sleep]:
        runtime.gopark(0x62a8e04f9d60?, 0x0?, 0x0?, 0x0?, 0x0?)
        	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000085f58 sp=0xc000085f38 pc=0x4863b6
        time.Sleep(0x6fc23ac00)
        	/usr/local/go/src/runtime/time.go:195 +0x135 fp=0xc000085f98 sp=0xc000085f58 pc=0x4b27f5
        sigs.k8s.io/controller-runtime/pkg/log.init.0.func1()
        	/root/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/log/log.go:63 +0x38 fp=0xc000085fe0 sp=0xc000085f98 pc=0x184def8
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc000085fe8 sp=0xc000085fe0 pc=0x4b5a01
        created by sigs.k8s.io/controller-runtime/pkg/log.init.0
        	/root/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/log/log.go:62 +0x25
        
        goroutine 51 [select]:
        runtime.gopark(0xc0000867a0?, 0x2?, 0x0?, 0x0?, 0xc000086764?)
        	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc000099de0 sp=0xc000099dc0 pc=0x4863b6
        runtime.selectgo(0xc000099fa0, 0xc000086760, 0x0?, 0x0, 0x0?, 0x1)
        	/usr/local/go/src/runtime/select.go:328 +0x7bc fp=0xc000099f20 sp=0xc000099de0 pc=0x495abc
        k8s.io/klog/v2.(*flushDaemon).run.func1()
        	/root/go/pkg/mod/k8s.io/klog/v2@v2.90.0/klog.go:1130 +0x11e fp=0xc000099fe0 sp=0xc000099f20 pc=0xadc35e
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc000099fe8 sp=0xc000099fe0 pc=0x4b5a01
        created by k8s.io/klog/v2.(*flushDaemon).run
        	/root/go/pkg/mod/k8s.io/klog/v2@v2.90.0/klog.go:1126 +0x17b
        
        goroutine 52 [select]:
        runtime.gopark(0xc001cc34d8?, 0x2?, 0x1f?, 0x93?, 0xc001cc341c?)
        	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc001cc3258 sp=0xc001cc3238 pc=0x4863b6
        runtime.selectgo(0xc001cc34d8, 0xc001cc3418, 0xd?, 0x0, 0x0?, 0x1)
        	/usr/local/go/src/runtime/select.go:328 +0x7bc fp=0xc001cc3398 sp=0xc001cc3258 pc=0x495abc
        net.(*Resolver).lookupIPAddr(0x3f6de20, {0x2cd4fb0?, 0xc0018cf620}, {0x28fdb9d, 0x3}, {0xc001463e50, 0x9})
        	/usr/local/go/src/net/lookup.go:327 +0x51b fp=0xc001cc35a8 sp=0xc001cc3398 pc=0x65277b
        net.(*Resolver).internetAddrList(0x2cd4fb0?, {0x2cd4fb0?, 0xc0018cf620?}, {0x28fdb9d, 0x3}, {0xc001463e50?, 0xf?})
        	/usr/local/go/src/net/ipsock.go:288 +0x67a fp=0xc001cc36c8 sp=0xc001cc35a8 pc=0x64fd7a
        net.(*Resolver).resolveAddrList(0x3fa9bc0?, {0x2cd4fb0, 0xc0018cf620}, {0x28fe4cd, 0x4}, {0x28fdb9d?, 0xc001463e50?}, {0xc001463e50, 0xf}, {0x0, ...})
        	/usr/local/go/src/net/dial.go:222 +0x41b fp=0xc001cc37d0 sp=0xc001cc36c8 pc=0x63705b
        net.(*Dialer).DialContext(0xc0018cf740, {0x2cd4fb0, 0xc0018cf620}, {0x28fdb9d, 0x3}, {0xc001463e50, 0xf})
        	/usr/local/go/src/net/dial.go:410 +0x448 fp=0xc001cc3920 sp=0xc001cc37d0 pc=0x637fe8
        net.(*Dialer).DialContext-fm({0x2cd4fb0?, 0xc0018cf620?}, {0x28fdb9d?, 0x9?}, {0xc001463e50?, 0x9?})
        	<autogenerated>:1 +0x5b fp=0xc001cc3968 sp=0xc001cc3920 pc=0x80ad7b
        net/http.(*Transport).dial(0xc00191cfc0?, {0x2cd4fb0?, 0xc0018cf620?}, {0x28fdb9d?, 0x0?}, {0xc001463e50?, 0x0?})
        	/usr/local/go/src/net/http/transport.go:1170 +0xda fp=0xc001cc39c0 sp=0xc001cc3968 pc=0x7f2e5a
        net/http.(*Transport).dialConn(0xc0018ea640, {0x2cd4fb0, 0xc0018cf620}, {{}, 0xc00191cfc0, {0xc001acbc80, 0x5}, {0xc001463e10, 0xe}, 0x0})
        	/usr/local/go/src/net/http/transport.go:1608 +0x83f fp=0xc001cc3ed8 sp=0xc001cc39c0 pc=0x7f61df
        net/http.(*Transport).dialConnFor(0x0?, 0xc0000f1970)
        	/usr/local/go/src/net/http/transport.go:1450 +0xb0 fp=0xc001cc3fc0 sp=0xc001cc3ed8 pc=0x7f4a70
        net/http.(*Transport).queueForDial.func1()
        	/usr/local/go/src/net/http/transport.go:1419 +0x2a fp=0xc001cc3fe0 sp=0xc001cc3fc0 pc=0x7f498a
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc001cc3fe8 sp=0xc001cc3fe0 pc=0x4b5a01
        created by net/http.(*Transport).queueForDial
        	/usr/local/go/src/net/http/transport.go:1419 +0x3f2
        
        goroutine 53 [select]:
        runtime.gopark(0xc001c85c28?, 0x2?, 0x0?, 0x30?, 0xc001c85c0c?)
        	/usr/local/go/src/runtime/proc.go:363 +0xd6 fp=0xc001c85a98 sp=0xc001c85a78 pc=0x4863b6
        runtime.selectgo(0xc001c85c28, 0xc001c85c08, 0xc001b4a270?, 0x0, 0x5414d4?, 0x1)
        	/usr/local/go/src/runtime/select.go:328 +0x7bc fp=0xc001c85bd8 sp=0xc001c85a98 pc=0x495abc
        net.cgoLookupIP({0x2cd4f40, 0xc001a52e80}, {0x28fdb9d, 0x3}, {0xc001463e50, 0x9})
        	/usr/local/go/src/net/cgo_unix.go:228 +0x1aa fp=0xc001c85cc8 sp=0xc001c85bd8 pc=0x66856a
        net.(*Resolver).lookupIP(0x3f6de20, {0x2cd4f40, 0xc001a52e80}, {0x28fdb9d, 0x3}, {0xc001463e50, 0x9})
        	/usr/local/go/src/net/lookup_unix.go:74 +0x128 fp=0xc001c85e18 sp=0xc001c85cc8 pc=0x6568e8
        net.(*Resolver).lookupIP-fm({0x2cd4f40?, 0xc001a52e80?}, {0x28fdb9d?, 0x0?}, {0xc001463e50?, 0x0?})
        	<autogenerated>:1 +0x5b fp=0xc001c85e60 sp=0xc001c85e18 pc=0x66cebb
        net.glob..func1({0x2cd4f40?, 0xc001a52e80?}, 0x0?, {0x28fdb9d?, 0x0?}, {0xc001463e50?, 0x7f4a70?})
        	/usr/local/go/src/net/hook.go:23 +0x3d fp=0xc001c85ea0 sp=0xc001c85e60 pc=0x64593d
        net.(*Resolver).lookupIPAddr.func1()
        	/usr/local/go/src/net/lookup.go:321 +0x9f fp=0xc001c85f38 sp=0xc001c85ea0 pc=0x65301f
        internal/singleflight.(*Group).doCall(0x3f6de30, 0xc001b00a00, {0xc001463e60, 0xd}, 0xc0000f1970?)
        	/usr/local/go/src/internal/singleflight/singleflight.go:95 +0x3b fp=0xc001c85fa8 sp=0xc001c85f38 pc=0x63329b
        internal/singleflight.(*Group).DoChan.func1()
        	/usr/local/go/src/internal/singleflight/singleflight.go:88 +0x36 fp=0xc001c85fe0 sp=0xc001c85fa8 pc=0x633236
        runtime.goexit()
        	/usr/local/go/src/runtime/asm_amd64.s:1594 +0x1 fp=0xc001c85fe8 sp=0xc001c85fe0 pc=0x4b5a01
        created by internal/singleflight.(*Group).DoChan
        	/usr/local/go/src/internal/singleflight/singleflight.go:88 +0x315
```

**Comment by apeabody** [comment]
> @apeabody Do you have some more context about this such as the OS and whether you are running inside a container? From the stack traces it looks like you are running on amd architecture. I've tried reproducing it on linux with amd, but I wasn't able to reproduce.

Hi @mortent - Let me follow-up with you directly.

**Comment by mortent** [comment]
So this is an interesting case. When we added support for wasm in kpt, we had to also start compiling kpt with cgo enabled. We initially distributed dynamically linked binaries, but as people reported issues with using kpt on images using alpine/distroless, we switched to using statically linked binaries.

As I looked into this issue, I realized that even with a statically linked binary, some system calls will still happen dynamically and `getaddrinfo` (as shown in your stacktrace) is one of those. As far as I can tell, glibc will always require that these calls happen dynamically as it depends on [NSS](https://en.wikipedia.org/wiki/Name_Service_Switch), which means kpt needs the same version of glibc available at runtime as it was compiled with. This seems to be the root of the panic you are seeing.

Golang supports the `netgo` tag to prevent go from using glibc from the `net` package and thereby requiring that the resulting binary is dynamically linked. We can build kpt with this tag and it would actually resolve the issue you found here. It is also possible to force go to use the pure-go version of the `net` package by setting the `GODEBUG=netdns=go` env variable before running kpt. This is described in more detail in https://mt165.co.uk/blog/static-link-go/.

However, if we build with the `netgo` tag, cgo will still print warnings, as wasmtime-go still uses these glibc functions (dlopen, getpwuid_r, getaddrinfo) that needs dynamic linking. And for the native wasm code, using the pure-go version is not an option.

So there doesn't seem to be a viable path to providing fully statically linked binaries of kpt with wasm support. Since wasm is still in alpha, I think we will disable it for the normal releases (using build tags) and for now users of kpt that want to try wasm will need to build from source. We can consider providing dynamically linked binaries of kpt for different unix distributions in the future or see if it possible to have statically linked binaries using musl.
